src/Symfony/Component/JsonEncoder/DependencyInjection/EncodablePass.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\JsonEncoder\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Sets the encodable classes to the services that need them.
+ *
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ */
+class EncodablePass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        if (!$container->hasDefinition('json_encoder.encoder')) {
+            return;
+        }
+
+        $encodableClassNames = [];
+
+        // retrieve concrete services tagged with "json_encoder.encodable" tag
+        foreach (array_keys($container->findTaggedServiceIds('json_encoder.encodable')) as $id) {
```

Comment:
array_keys is overhead

why don't you call removeDefinition anymore?

Reply:
I wasn't sure that removing a user land defined service was a good idea.

But after a second thought, encodable classes should not really be considered as services, so I put the `removeDefinition` back, thanks.